app/database.js

```diff
@@ -0,0 +1,201 @@
+import { MongoClient, ServerApiVersion } from 'mongodb'
+import _ from 'lodash'
+
+class Database {
+	static normalizeName (name) {
+		return name.toLowerCase()
+	}
+
+	constructor ({
+		connectionURI = process.env.WIDSCHIBOT_DB_URI,
+		dbName = process.env.WIDSCHIBOT_DB_NAME,
+		dbOptions,
+	} = {}) {
+		if (!connectionURI) {
+			throw new Error('Connecting to MongoDB: No connection URI specified as argument or WIDSCHIBOT_DB_URI environment variable.')
+		}
+		if (!dbName) {
+			throw new Error('Connecting to MongoDB: No connection URI specified as argument or WIDSCHIBOT_DB_NAME environment variable.')
+		}
+		dbOptions = {
+			serverApi: ServerApiVersion.v1,
+			...dbOptions,
+		}
+
+		this.dbName = dbName
+		this.connectionURI = connectionURI
+		this.collectionName = 'users'
+		this.client = new MongoClient(connectionURI, dbOptions)
+	}
+
+	async connect () {
+		await this.client.connect()
+		this.db = this.client.db(this.dbName)
+		this.collection = this.db.collection(this.collectionName)
+	}
+
+	async disconnect () {
+		await this.client.close()
+	}
+
+	async clear () {
+		await this.collection.deleteMany({})
+	}
+
+	async addUser (chatID, userID) {
+		const cursor = await this.collection.aggregate([
+			{
+				$match: { chatID: chatID },
+			}, {
+				$group: {
+					_id: null,
+					average: { $avg: '$score' },
+				},
+			},
+		])
+		let average = 0
+		// eslint-disable-next-line no-unreachable-loop
+		for await (const first of cursor) {
+			average = first.average || 0
+			break
+		}
+		await cursor.close() // We should close the cursor manually since we might not have exhausted it.
+
+		userID = Database.normalizeName(userID)
+		const newUser = {
+			chatID: chatID,
+			userID: userID,
+			vacation: false,
+			score: average,
+		}
+
+		const result = await this.collection.findOneAndUpdate({
+			chatID: chatID,
+			userID: userID,
+		}, {
+			$setOnInsert: newUser,
+		}, {
+			upsert: true,
+		})
+		if (result.value === null) {
+			return newUser
+		} else {
+			return null
+		}
```

Comment:
_optional_ this would probably be concidered more concise:
```suggestion
		return (result.value === null) ? newUser : null
```